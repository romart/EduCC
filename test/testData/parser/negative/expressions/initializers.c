
struct S { float x; int *ptr; int *ptr2; };
struct S s1 = { { { (void*)0, 01.f, 5 } , 0 }, { 0, 0.1f, "ccc" } , 0 };
int a1[20];
int a2[] = a1;
int *pa = a1;
int a3[];
int a4[20] = 0;
int b[][20] = {} ;
struct S s2 = { 0 };
int i = 10;
float f = pa;
float f2 = 1.0f;
int i2 = f2;
int i3 = s2;
void x;
int a5[]={ { { 0, 0}, 0, { 0, 0}}, { 0}, {0, { }}}
