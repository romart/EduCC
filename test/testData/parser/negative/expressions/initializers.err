test/testData/parser/negative/expressions/initializers.c:21:3: error: assigning to 'float' from incompatible type '*void'
test/testData/parser/negative/expressions/initializers.c:12:5: error: invalid initializer
test/testData/parser/negative/expressions/initializers.c:1:7: error: definition of variable with array type needs an explicit size or an initializer
test/testData/parser/negative/expressions/initializers.c:14:8: error: array type 'signed int[20]' is not assignable
test/testData/parser/negative/expressions/initializers.c:11:12: error: assigning to 'float' from incompatible type '*signed int'
test/testData/parser/negative/expressions/initializers.c:10:15: error: assigning to 'signed int' from incompatible type 'STRUCT S'
test/testData/parser/negative/expressions/initializers.c:1:16: error: variable has incomplete type 'void'
