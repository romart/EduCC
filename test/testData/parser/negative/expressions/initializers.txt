FILE test/testData/parser/negative/expressions/initializers.c
  STRUCT S
    float x #0
    *signed int ptr #4
    *signed int ptr2 #12
  STRUCT_END
----
  STRUCT SX
    signed int a #0
    STRUCT S s #4
    STRUCT S[2] as #28
  STRUCT_END
----
  STRUCT S s0
----
  STRUCT S s1 = \
    INIT_BEGIN
      1.000000
      (*signed int)0
      (*signed int)0
    INIT_END
----
  STRUCT S s2 = \
    INIT_BEGIN
      1.000000
      (*signed int)0
      (*signed int)0
    INIT_END
----
  STRUCT S s3 = \
    INIT_BEGIN
      1.000000
      (*signed int)0
      (*signed int)0
    INIT_END
----
  STRUCT S s4 = \
    INIT_BEGIN
      1.000000
      (*signed int)0
      (*signed int)0
    INIT_END
----
  STRUCT S[3] s5 = \
    INIT_BEGIN
      INIT_BEGIN
        1.000000
        (*signed int)0
        (*signed int)0
      INIT_END
      INIT_BEGIN
        1.000000
        (*signed int)0
        (*signed int)0
      INIT_END
      INIT_BEGIN
        1.000000
        (*signed int)0
        (*signed int)0
      INIT_END
    INIT_END
----
  STRUCT S s6 = \
    INIT_BEGIN
      1.000000
      (*signed int)0
      (*signed int)0
    INIT_END
----
  STRUCT S s7 = \
    INIT_BEGIN
      (*void)0
      1.000000
      ERROR EXPR
    INIT_END
----
  STRUCT S[3] s8 = \
    INIT_BEGIN
      INIT_BEGIN
        (*void)0
        1.000000
        5
      INIT_END
      INIT_BEGIN
        0
        0.100000
        "ccc"
      INIT_END
      0
    INIT_END
----
  STRUCT S[1] sa = \
    INIT_BEGIN
      0
    INIT_END
----
  STRUCT SX sx1 = \
    INIT_BEGIN
      0
      *s0
      INIT_BEGIN
        0
      INIT_END
    INIT_END
----
  STRUCT S[3] as1 = \
    INIT_BEGIN
      0
      0
      0
    INIT_END
----
  STRUCT S[3] as2 = \
    INIT_BEGIN
      INIT_BEGIN
        0
      INIT_END
      INIT_BEGIN
        0
      INIT_END
      INIT_BEGIN
        0
      INIT_END
    INIT_END
----
  signed int[4] a = \
    INIT_BEGIN
      0
      1
      2
      3
      4
      5
      6
      7
      ERROR EXPR
      8
      9
    INIT_END
----
  (signed int[2])[3] b = \
    INIT_BEGIN
      INIT_BEGIN
        0
        1
      INIT_END
      INIT_BEGIN
        2
        3
        4
        5
        6
        7
        ERROR EXPR
        8
      INIT_END
      INIT_BEGIN
        9
      INIT_END
    INIT_END
----
  signed int[9] c = \
    INIT_BEGIN
      0
      1
      3
      4
      5
      6
      7
      8
      9
    INIT_END
----
  signed int[9] d = \
    INIT_BEGIN
      0
      1
      3
      4
      5
      6
      7
      8
      9
    INIT_END
----
  signed int[9] e = \
    INIT_BEGIN
      0
      1
      3
      4
      5
      6
      7
      8
      9
    INIT_END
----
  signed int[9] f = \
    INIT_BEGIN
      0
      1
      3
      4
      5
      6
      7
      8
      9
    INIT_END
----
  signed int[4] g = \
    INIT_BEGIN
      0
      1
      2
      3
    INIT_END
----
  signed int[4] h = \
    INIT_BEGIN
      0
      1
      2
      3
      4
    INIT_END
----
  signed int[4] i = \
    INIT_BEGIN
      0
      1
      2
      3
      4
      ERROR EXPR
    INIT_END
----
  signed int[20] a1
----
  *signed int pa = \
    a1
----
  float f = \
    *pa
----
  signed int[-1] a3
----
  signed int[20] a4 = \
    0
----
  (signed int[20])[] b2 = \
    INIT_BEGIN

    INIT_END
----
  STRUCT S s22 = \
    INIT_BEGIN
      0
    INIT_END
----
  signed int i2 = \
    10
----
  float f2 = \
    1.000000
----
  ERROR TYPE x
----
  signed int[3] a5 = \
    INIT_BEGIN
      0
      0
      0
      0
      0
      0
      0
      ERROR EXPR
    INIT_END
----
  signed int aa = \
    INIT_BEGIN

    INIT_END
----
  (*signed int)[2] ap = \
    INIT_BEGIN
      1.000000
      2.000000
    INIT_END
----
  float[2] af = \
    INIT_BEGIN
      (*signed int)0
      (*signed int)1
    INIT_END
----
  FUN signed int func 
  BEGIN
    signed int[-1] a2 = \
      a1
    *signed int pa = \
      a1
    float f = \
      *pa
    signed int i2 = \
      *f2
    signed int i3 = \
      *s2
    STRUCT SX sx2 = \
      INIT_BEGIN
        0
        *s0
        INIT_BEGIN
          0
        INIT_END
      INIT_END    
  END
----
  E signed int eei = \
    20
