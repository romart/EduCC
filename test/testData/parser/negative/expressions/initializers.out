test/testData/parser/negative/expressions/initializers.c:31:3: warning: excess elements in scalar initializer
test/testData/parser/negative/expressions/initializers.c:43:3: warning: excess elements in scalar initializer
test/testData/parser/negative/expressions/initializers.c:50:3: warning: incompatible integer to pointer conversion assigning to '*signed int' from 'C signed long'
test/testData/parser/negative/expressions/initializers.c:53:3: warning: excess elements in scalar initializer
test/testData/parser/negative/expressions/initializers.c:69:3: warning: incompatible integer to pointer conversion assigning to '*signed int' from 'C signed long'
test/testData/parser/negative/expressions/initializers.c:19:17: warning: excess elements in scalar initializer
test/testData/parser/negative/expressions/initializers.c:23:17: warning: excess elements in scalar initializer
test/testData/parser/negative/expressions/initializers.c:46:17: warning: excess elements in scalar initializer
